Port vertical changes

prashanth-intel opened the pull request.

YadongQi commented on scripts/setup_civ_ini.sh:
```diff
@@ -0,0 +1,44 @@
+CIV_WORK_DIR=$(pwd)
+USER_HOME=$(getent passwd $SUDO_USER | cut -d: -f6)
+
+function setup_civ_ini() {
```
I think its better to keep this script file in vertical.

YadongQi commented:
The S branch has been into maintaince status, features are not accepted in genneral unless there is a strong requirement. Please make sure you really want these new updates.
And please also fix the code scaning issues.

prashanth-intel commented:
> The S branch has been into maintaince status, features are not accepted in genneral unless there is a strong requirement. Please make sure you really want these new updates. And please also fix the code scaning issues.

Hi Yadong. Removed the commit for ini file. Fixed all relevant code scanning issues. There is one issue which does not seem critical as I am appending an integer and not string.

YadongQi commented on scripts/setup_host.sh:
```diff
@@ -667,6 +667,8 @@ ubu_enable_host_sriov
 
 install_vm_manager
+create_aaf_dir
+setup_civ_ini
```
calling inexistent function?

YadongQi commented:
> > The S branch has been into maintaince status, features are not accepted in genneral unless there is a strong requirement. Please make sure you really want these new updates. And please also fix the code scaning issues.
> 
> Hi Yadong. Removed the commit for ini file. Fixed all relevant code scanning issues. There is one issue which does not seem critical as I am appending an integer and not string.

Suggest to use strncat() to avoid unsafe use of strcat().

prashanth-intel commented:
> > > The S branch has been into maintaince status, features are not accepted in genneral unless there is a strong requirement. Please make sure you really want these new updates. And please also fix the code scaning issues.
> > 
> > 
> > Hi Yadong. Removed the commit for ini file. Fixed all relevant code scanning issues. There is one issue which does not seem critical as I am appending an integer and not string.
> 
> Suggest to use strncat() to avoid unsafe use of strcat().

Done. Thanks for the suggestion.

YadongQi commented on src/guest/start.c:
```diff
@@ -856,6 +908,25 @@ static void strip_duplicate(gchar *val, const gchar *inner_cmd)
+		char *num;
+		asprintf(&num, "%d", avail);
+		strcpy(qmp_pwr_socket, "/tmp/qmp-pwr-socket-");
+		if(sizeof(qmp_pwr_socket) > 0) {
+			strncat(qmp_pwr_socket, num, sizeof(int));
+		}
```
sizeof(qmp_pwr_socket) always > 0, so the condition always true. I think you mean to:
```
if (strlen(num) < (sizeof(qmp_pwr_socket) - strlen(qmp_pwr_socket)) {
    strncat(qmp_pwr_socket, num, (sizeof(qmp_pwr_socket) - strlen(qmp_pwr_socket));
}
```

YadongQi commented on src/guest/start.c:
```diff
@@ -906,6 +977,11 @@ int start_guest(char *name)
 	cx = snprintf(p, size, " -qmp unix:%s/.%s"CIV_GUEST_QMP_SUFFIX",server,nowait", civ_config_path, qname);
 	p += cx; size -= cx;
 
+	const char qmp_sock[MAX_CMDLINE_LEN] = { 0 };
```
remove `const ` qualifier.

YadongQi commented on src/guest/start.c:
```diff
@@ -856,6 +908,25 @@ static void strip_duplicate(gchar *val, const gchar *inner_cmd)
 		g_free(to_split);
 }
 
+void set_pwr_ctrl(char * qmp_cmd) {
```
add `static` qualifier for this function.

YadongQi commented on src/guest/start.c:
```diff
@@ -552,6 +596,13 @@ static void cleanup_passthrough(void) {
 	}
 }
 
+void cleanup_pwr_ctrl() {
```
add `static` qualifier for this function.

sysopenci reviewed (commented):
Failure when executing preintegration builder

FAILURE:CheckBug Bad comments/Bugs

More details, logs, and binaries: /absp/builders/celadon_smr0_stable-preintegration/builds/283

sysopenci reviewed (commented):
Failure when executing preintegration builder

FAILURE:CheckBug Bad comments/Bugs

More details, logs, and binaries: /absp/builders/celadon_smr0_stable-preintegration/builds/284

sysopenci reviewed (commented):
Failure when executing preintegration builder

FAILURE:Update Bugs And Epics Update Bugs And Epics

More details, logs, and binaries: /absp/builders/celadon_smr0_stable-preintegration/builds/287

sysopenci reviewed (commented):
Failure when executing preintegration builder

FAILURE:CheckBug Bad comments/Bugs

More details, logs, and binaries: /absp/builders/celadon_smr0_stable-preintegration/builds/290

sysopenci reviewed (commented):
Failure when executing preintegration builder

FAILURE:Update Bugs And Epics Update Bugs And Epics

More details, logs, and binaries: /absp/builders/celadon_smr0_stable-preintegration/builds/291

sysopenci reviewed (commented):
Failure when executing preintegration builder

FAILURE:CheckApprovals Some checks are failed. See the step results.html

More details, logs, and binaries: /absp/builders/celadon_smr0_stable-preintegration/builds/292

sysopenci reviewed (commented):
Failure when executing preintegration builder

FAILURE:CheckBug Bad comments/Bugs

More details, logs, and binaries: /absp/builders/celadon_smr0_stable-preintegration/builds/293

sysopenci reviewed (commented):
Failure when executing preintegration builder

FAILURE:Trigger Builds Triggered: build-cels - r1 (caas), build-cels - r1 (caas_sriov), build-cels - r2 (r2_caas), build-cels - r2 (r2_caas_sriov) - extra

More details, logs, and binaries: /absp/builders/celadon_smr0_stable-preintegration/builds/294

sysopenci reviewed (commented):
Failure when executing preintegration builder

FAILURE:Trigger Builds cancelled

More details, logs, and binaries: /absp/builders/celadon_smr0_stable-preintegration/builds/296

sysopenci reviewed (commented):
Failure when executing preintegration builder

FAILURE:Trigger Builds cancelled

More details, logs, and binaries: /absp/builders/celadon_smr0_stable-preintegration/builds/299

sysopenci reviewed (commented):
PR integrated into 'preintegration' builder



More details, logs, and binaries: /absp/builders/celadon_smr0_stable-preintegration/builds/297

Done by "Suresh, Prashanth" <prashanth.suresh@intel.com>

sysopenci reviewed (commented):
cactus finished building and testing your changes 

It was requested for merge by "Suresh, Prashanth" <prashanth.suresh@intel.com> 

so we merge it. Thanks for your contribution.

more details, logs, and binaries: /absp/builders/celadon_smr0_stable-mergerequest/builds/134
